### PR TITLE
Remove a couple type ignores + cleanup CacheMappingView

### PR DESCRIPTION
**Pull request by A5rocks**
### Summary
I really wanted to remove a few of those pesky type ignores, you know? But I may have added more than I removed... Anyways, this PR also contains cache changes as when I asked around about whether I could change some interfaces, I got directed to remove some things instead. 

### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [ ] I have made unittests according to the code I have added/modified/deleted.

### Related issues
None.

### CODE REVIEW

File: hikari/files.py
Comment (A5rocks) [inline hikari/files.py:956]: I'm not *quite* sure these `isinstance` checks are legal here, but I'll check soon:tm: if they actually are.

File: hikari/internal/cache.py
Comment (FasterSpeeding) [inline hikari/internal/cache.py:728]: might as well just do a generative list literal rather than using map here

File: hikari/internal/cache.py
Comment (FasterSpeeding) [inline hikari/internal/cache.py:161]: probably want to update the typing for this method on hikari.api.cache.CacheView and the few other classes which inherit from it (plus you forgot the `, /`)

File: hikari/internal/cache.py
Comment (FasterSpeeding) [inline hikari/internal/cache.py:107]: you can just not include builder in this overload

File: hikari/internal/cache.py
Comment (davfsa) [inline hikari/internal/cache.py:100]:
```suggestion
    __slots__: typing.Sequence[str] = ("_builder", "_data")
```

File: hikari/files.py
Comment (davfsa) [inline hikari/files.py:970]: How do these checks differentiate from `inspect.isgenerator` speed wise?

File: hikari/api/cache.py
Comment (A5rocks) [inline hikari/api/cache.py:61]: Does anyone happen to know whether this is valid? I mean, it passes type checking and the tests don't error but... it just kind of feels wrong.

Comment (davfsa) [reply hikari/api/cache.py:61]: We have been doing it already. I don't know what the correct way of doing it is, but we have using this way already. https://github.com/hikari-py/hikari/blob/master/hikari/api/rest.py#L496-L587

File: hikari/internal/aio.py
Comment (davfsa) [inline hikari/internal/aio.py:40]: `mypy` is not installed as a requirement as hikari, so this will break if people try to analyze typehints without mypy installed

File: hikari/config.py
Comment (davfsa) [inline hikari/config.py:337]: This is not correct, as we want to make sure that `_ssl` cannot hold a bool. Unfortunately, due to how attr works, the typehint we provide will be the same as the one used for both the `__init__` and the value hold. This type ignore is the only way we can fix this

Comment (A5rocks) [reply hikari/config.py:337]: Are you sure? Because `ssl` has the docstring:
https://github.com/hikari-py/hikari/blob/15624037e97d989c2e22b1bb1e1f8f6afd28f8ac/hikari/config.py#L341-L376

Which mentions it can be assigned either a bool or an SSLContext,

Comment (davfsa) [reply hikari/config.py:337]: Yes, we allow passing both bool and SSLContext. A bool will later be converted into a SSLContext in `_ssl_factory`. The validator just asserts that what was returned was an SSLContext

Comment (A5rocks) [reply hikari/config.py:337]: Shouldn't `_ssl` be `ssl_.SSLContext` then? mypy at the very least takes the type of the converter (`_ssl_factory`) for the type in `__init__`, not the type hint: https://github.com/python/mypy/blob/master/mypy/plugins/attrs.py#L113

File: hikari/config.py
Comment (davfsa) [inline hikari/config.py:337]: Since we are here, could you merge this with the `ssl` property. There is no need for it anymore :)

Comment (A5rocks) [reply hikari/config.py:337]: done

File: hikari/internal/aio.py
Comment (A5rocks) [reply hikari/internal/aio.py:41]: I think this is probably fine -- if anyone complains, it can easily be fixed then. Adding a runtime dependency for something that is supposed to be static just doesn't sit right with me...

Additionally, this will only error if forward annotations suddenly die, or if someone runs `typing.get_type_hints` on the functions in question: both are not very likely imo, especially this is internal to the library.

**Comment by A5rocks** [comment]
This PR needs:

> First-time contributors need a maintainer to approve running workflows

**Comment by A5rocks** [comment]
Gonna close and re-open to give github actions the little nudge it needs I guess?

File: hikari/internal/aio.py
Comment (davfsa) [reply hikari/internal/aio.py:41]: @FasterSpeeding ?

Comment (FasterSpeeding) [reply hikari/internal/aio.py:41]: Seems fine to me

**Comment by A5rocks** [comment]
@FasterSpeeding you may want to review the latest commit, as it changes some of the cache things.

**Review by davfsa — APPROVED** [review]
Would like to get Snabs approval before merging :)

File: hikari/internal/cache.py
Comment (davfsa) [reply hikari/internal/cache.py:117]: Ah, my bad. Rocks, can you please revert?

File: hikari/config.py
Comment (A5rocks) [reply hikari/config.py:264]: Pretty sure attrs doesn't add `attr.setters.convert` by default, I'll check locally.

Comment (A5rocks) [reply hikari/config.py:264]: Confirmed:
```py
In [1]: import attr

In [2]: @attr.define()
   ...: class Example:
   ...:     b: int = attr.ib(converter=int)
   ...: 

In [3]: n = Example(5)

In [4]: n.b = '42'

In [5]: n
Out[5]: Example(b='42')

In [6]: @attr.define(on_setattr=[attr.setters.convert, attr.setters.validate])
   ...: class Example:
   ...:     b: int = attr.ib(converter=int)
   ...: 

In [7]: nn = Example(5)

In [8]: nn.b = '42'

In [9]: nn
Out[9]: Example(b=42)
```

File: hikari/internal/cache.py
Comment (A5rocks) [reply hikari/internal/cache.py:117]: Done.

File: hikari/config.py
Comment (FasterSpeeding) [reply hikari/config.py:264]: Welp that's not the behaviour I'm seeing, on the latest ver of attr i'm getting
```
ok 123
<class 'int'>
```
from the following code
```py
import attr

def test_convert(value: str) -> int:
    print("ok", value)
    return int(value)

@attr.define()
class Foo:
    foo = attr.field(converter=test_convert)


print(type(Foo("123").foo))
```

Also under current behaviour the _ssl field is using a converter without setting the on_setattr and validation or the property's assertion statement would be constantly erroring if the converter wasn't running and it was being left as a boolean value

Comment (A5rocks) [reply hikari/config.py:264]: Yeah, converters always work in the initializer. But, the docstring mentions `If assigned to <boolean value>`, which has the connotation of `settings.ssl = False` after making the settings. I may be misreading though!

File: hikari/internal/cache.py
Comment (davfsa) [inline hikari/internal/cache.py:174]: You reverted this one, but I think it was better the other way

File: hikari/config.py
Comment (FasterSpeeding) [reply hikari/config.py:264]: Well typing wise you can't assigned a bool to it anymore cause as far as MYPY's concerned since it's typed as as SSLContext it therefore can only be set as an SSLContext attribute wise although currently this is an unsettable property so that whole "assigned" thing was probably just ambiguous phrasing or outdated docs

Comment (FasterSpeeding) [reply hikari/config.py:264]: Also it's probably preferable that people aren't encouraged to set those fields like that since there's no guarantee it'll update the behaviour of whatever client it's attached to so those docs should prob be updated ~~and there might actually be merit in freezing these classes~~

Comment (FasterSpeeding) [reply hikari/config.py:264]: > probably just ambiguous phrasing or outdated docs
It's the latter

Comment (A5rocks) [reply hikari/config.py]: Done

File: hikari/internal/cache.py
Comment (A5rocks) [reply hikari/internal/cache.py]: Done

**Review by davfsa — DISMISSED** [review]

File: hikari/internal/cache.py
Comment (FasterSpeeding) [inline hikari/internal/cache.py:139]: this else statement is redundant

**Review by davfsa — APPROVED** [review]

**Review by FasterSpeeding — APPROVED** [review]

**Pull request merged**
